index.js

```diff
@@ -2,15 +2,26 @@
 
 const fp = require('fastify-plugin')
 const { Cache } = require('async-cache-dedupe')
+const S = require('fluent-json-schema')
+const Ajv = require('ajv')
+const ajv = new Ajv()
 
 module.exports = fp(async function (app, { all, policy, ttl, cacheSize, skip, storage, onHit, onMiss, onSkip, ...other }) {
```

Comment:
let's consider removing the destructuring so we can provide the whole set of options to the validation